capstone-project/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -76,28 +77,45 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         Action action = Action.values()[actionNum];
         Gson gson = new Gson();
         long markerId;
+        String userToken = request.getParameter("userToken");
+        Optional<String> userId = verifyToken(userToken);
         ServletContext context = getServletContext();
         switch (action) {
             case CREATE:
-                String userToken = request.getParameter("userToken");
-                Optional<String> userId = verifyToken(userToken);
                 Marker newMarker = gson.fromJson(request.getParameter("marker"), Marker.class);
+                HashMap<String, String> returnParameters = new HashMap<>();
                 newMarker.setUserId(userId);
                 markerId = storeMarker(newMarker);
                 // The ID of the entity need to be updated in the FE as well
-                response.getWriter().println(markerId);
+                returnParameters.put("id", Long.toString(markerId));
+                if (userId.isPresent()){
+                    returnParameters.put("userId", userId.get());
+                }
+                response.getWriter().println(gson.toJson(returnParameters));
                 break;
             case UPDATE:
                 Marker updatedMarker = gson.fromJson(request.getParameter("marker"), Marker.class);
                 try {
-                    updateMarker(updatedMarker);
-                } catch (EntityNotFoundException e) {
-                    context.log("Update failed, Entity not found. Error details: ", e);
+                    updateMarker(updatedMarker, userId);
+                } catch (EntityNotFoundException entityNotFoundException) {
```

Comment:
You can just catch the two exception types:

catch(EntityNotFoundException | GeneralSecurityException e) {
 context.log("Update failed. Error details: ", e);
}

The log will contain the exception type